### PR TITLE
Language detection and a patch

### PR DESCRIPTION
**Pull request by Teklad**
This resolves issue #281 in full.  It tries to detect default system locale by default, and if the translation for that locale doesn't exist defaults back to English. 

I'll need a couple of people to test this to make sure the patch @david-geiger sent in doesn't cause any issues for anyone else.  I used `export LANG=fr_CA.UTF-8` to test and it seemed to work okay on my end.

### CODE REVIEW

File: src/ui/include/nqqsettings.h [hunk 1]
```diff
@@ -63,6 +63,7 @@ class NqqSettings {
 public:
 
     BEGIN_GENERAL_CATEGORY(General)
+        NQQ_SETTING(FirstRun,                       bool,       true)
         NQQ_SETTING(Localization,                   QString,    "en")
```
Comment (danieleds) [inline src/ui/include/nqqsettings.h]: Instead of a "FirstRun" setting, can't you change the default value for "Localization" to an empty string and check if that is empty?

Comment (Teklad) [inline src/ui/include/nqqsettings.h]: Yea, but I was thinking about other settings that might need to be initialized in the future on a first run basis... if you don't think this is an issue I can default it to empty instead.

**Comment by Teklad** [comment]
I think that should cover your request.  No point in trying to think too far ahead here, lol.

File: src/ui/main.cpp [hunk 2]
```diff
@@ -60,10 +70,12 @@ int main(int argc, char *argv[])
     } else if (translator.load(QLocale(langCode),
                                QString("%1").arg(qApp->applicationName().toLower()),
                                QString("_"),
-                               QString("%1/../../share/%2/translations")
+                               QString("%1/../share/%2/translations")
```
Comment (danieleds) [inline src/ui/main.cpp]: Why this change? When installed, notepadqq is located at `/usr/lib/notepadqq/notepadqq-bin`, so the double `../` was ok.
Maybe add a comment next to that line, saying `// The Notepadqq binary gets installed at /usr/lib/notepadqq/notepadqq-bin`

Comment (Teklad) [inline src/ui/main.cpp]: You're right... this actually breaks the installation on my end (when installed)...  But according to #281 its already broken for another user, though I'm sure there's special circumstances involved.

I'll go ahead and revert this small change so it'll just be a feature request rather than a bug fix, but we need to see about a more reliable way to detect this once installed on the system in the future.

**Comment by danieleds** [comment]
Merged: 0a40d97edf71c505074210290b0291f3d6d3f72e
